src/main/java/com/fasterxml/jackson/core/io/IOContext.java

```diff
@@ -129,6 +136,8 @@ public class IOContext
      * @param managedResource Whether input source is managed (owned) by Jackson library
      *
      * @since 2.16
+     * @deprecated Since 2.16, use {@link #IOContext(StreamReadConstraints, StreamWriteConstraints, BufferRecycler, 
+     * ContentReference, boolean, ErrorReportConfiguration)} instead.
      */
     public IOContext(StreamReadConstraints src, StreamWriteConstraints swc, BufferRecycler br,
```

Comment:
Could you also add `@Deprecated` here (for some reason Eclipse warns if that's missing when there's Javadoc `@deprecated`).